### PR TITLE
simplify processing code

### PR DESCRIPTION
**Pull request by jamestwebber**
I was using this as a template for trying out BUStools. I found some of the processing code a bit hard to parse so I rewrote it to (I hope) be easier to understand. This version is a bit faster as well.

### CODE REVIEW

**Review by pmelsted — APPROVED** [review]
Nice simplification of the code

**Comment by jamestwebber** [comment]
I had a related question: is [this line](https://github.com/BUStools/Notebooks/blob/master/Notebooks/10x/v2/GeneCounts.ipynb?short_path=8ab5e83#L165) correct?

```python
for barcode in cell_gene:
    cg = cell_gene[barcode]
    s = len([cg[g] for g in cg]) # <<<
    barcode_hist[barcode] += s
```

As far as I can tell it's equivalent to `len(cg[g])` with an unnecessary list comprehension thrown in. Should `s` be the number of genes for that barcode or the total counts for that barcode? Looking at [line 244](https://github.com/BUStools/Notebooks/blob/master/Notebooks/10x/v2/GeneCounts.ipynb?short_path=8ab5e83#L244) it seems like maybe the latter was intended?
